From 665819a77801c59b7ae847442ddedbfc4031e189 Mon Sep 17 00:00:00 2001
From: Yann Gautier <yann.gautier@st.com>
Date: Wed, 24 Nov 2021 18:32:35 +0100
Subject: [PATCH 026/841] feat(st): manage ROM code boot_action field

Add a function to get the ROM code boot field boot_action. This allows
determining if this is a cold boot or a return from Standby.

Upstream-Status: Inappropriate [ST downstream https://github.com/STMicroelectronics/arm-trusted-firmware.git 5b1422d0de9620d4b632f36745eefe809d8276b0 feat(st): manage ROM code boot_action field]
Change-Id: I04100a72483b44655fe90386090bae69ad06ebe3
Signed-off-by: Yann Gautier <yann.gautier@st.com>
---
 plat/st/common/include/stm32mp_common.h | 1 +
 plat/st/common/stm32mp_common.c         | 7 +++++++
 2 files changed, 8 insertions(+)

diff --git a/plat/st/common/include/stm32mp_common.h b/plat/st/common/include/stm32mp_common.h
index 606da9743..3d436934a 100644
--- a/plat/st/common/include/stm32mp_common.h
+++ b/plat/st/common/include/stm32mp_common.h
@@ -21,6 +21,7 @@
 void stm32mp_save_boot_ctx_address(uintptr_t address);
 uintptr_t stm32mp_get_boot_ctx_address(void);
 uint16_t stm32mp_get_boot_itf_selected(void);
+uint32_t stm32mp_get_boot_action(void);
 
 bool stm32mp_is_single_core(void);
 bool stm32mp_is_closed_device(void);
diff --git a/plat/st/common/stm32mp_common.c b/plat/st/common/stm32mp_common.c
index fc32356fd..aeb9da0d5 100644
--- a/plat/st/common/stm32mp_common.c
+++ b/plat/st/common/stm32mp_common.c
@@ -71,6 +71,7 @@ unsigned int plat_get_syscnt_freq2(void)
 
 static uintptr_t boot_ctx_address;
 static uint16_t boot_itf_selected;
+static uint32_t boot_action_saved;
 
 void stm32mp_save_boot_ctx_address(uintptr_t address)
 {
@@ -78,6 +79,7 @@ void stm32mp_save_boot_ctx_address(uintptr_t address)
 
 	boot_ctx_address = address;
 	boot_itf_selected = boot_context->boot_interface_selected;
+	boot_action_saved = boot_context->boot_action;
 }
 
 uintptr_t stm32mp_get_boot_ctx_address(void)
@@ -90,6 +92,11 @@ uint16_t stm32mp_get_boot_itf_selected(void)
 	return boot_itf_selected;
 }
 
+uint32_t stm32mp_get_boot_action(void)
+{
+	return boot_action_saved;
+}
+
 uintptr_t stm32mp_ddrctrl_base(void)
 {
 	return DDRCTRL_BASE;
-- 
2.47.2

